10 спринт. Начало разработки приложения по оценки фильмов с использов…

EvgenyAlexandrovichM opened the pull request:
10 спринт. Начало разработки приложения по оценки фильмов с использованием Spring boot.
1 коммит

EvgenyAlexandrovichM committed:
10 спринт. Начало разработки приложения по оценки фильмов с использованием Spring boot.
1 коммит

EvgenyAlexandrovichM committed:
10 спринт. Начало разработки приложения по оценки фильмов с использованием Spring boot.
1 коммит

EvgenyAlexandrovichM committed:
10 спринт. Начало разработки приложения по оценки фильмов с использованием Spring boot.
1 коммит

EvgenyAlexandrovichM committed:
10 спринт. Начало разработки приложения по оценки фильмов с использованием Spring boot.
1 коммит

EvgenyAlexandrovichM committed:
10 спринт. Начало разработки приложения по оценки фильмов с использованием Spring boot.
1 коммит

EvgenyAlexandrovichM committed:
10 спринт. Начало разработки приложения по оценки фильмов с использованием Spring boot.
1 коммит

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,65 @@
+            films.put(film.getId(), film);
+            log.info("Создан фильм с id: {}", film.getId());
+            return new ResponseEntity<>(film, HttpStatus.CREATED);
+        } catch (Exception e) {
```
лучше ответ с ошибкой вынести в класс ValidationExceptionHandler и назвать его немножко иначе

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,65 @@
+        try {
+            if (film.getId() == null || !films.containsKey(film.getId())) {
+                log.warn("Фильм с id {} не найден", film.getId());
+                return new ResponseEntity<>(film, HttpStatus.NOT_FOUND);
```
здесь тоже лучше приготовить исключение, что сущность не найдена, бросить ее, а в ValidationExceptionHandler уже ответить. Т.е. в контроллере оставить только успешные ответы по существу

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,70 @@
+@RestController
+@RequestMapping("/users")
+@Slf4j
+public class UserController {
```
в этом классе тоже нужно вынести все ответы с ошибками заменить на выброс исключения, которое обработает ValidationExceptionHandler

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,70 @@
+    public ResponseEntity<User> create(@Valid @RequestBody User user) {
+        log.info("Запрос на создание пользователя: {}", user);
+        try {
+            if (user.getName() == null || user.getName().isEmpty()) {
```
эта логика нужна еще и в методе обновления, поэтому вынеси в отдельный метод и используй

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -1,12 +1,25 @@
+    @ValidFilmReleaseDate
+    private LocalDate releaseDate;
+    @Positive(message = "Продолжительность фильма должна быть положительным числом.")
+    private Integer duration;
```
сюда еще нужна аннотация @NotNull с комментарием

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/validators/FilmReleaseDateValidator.java:
```diff
@@ -0,0 +1,23 @@
+        if (releaseDate == null) {
+            return true;
+        }
+        return !releaseDate.isBefore(MIN_RELEASE_DATE);
```
чтобы не конвертировать результат releaseDate.isBefore(), что ухудшает читаемость кода, можно воспользоваться методом isAfter()

EvgenyAlexandrovichM committed:
10 спринт. 1 Ревью.
Исправлены недочеты по ревью + рефакторинг с использованием принципа single responsibility. Контроллеры больше не имеют логики обработки данных, а только отвечают на запросы, вся логика перенесена в сервис.

EvgenyAlexandrovichM committed:
10 спринт. 1 Ревью.
Исправлены недочеты по ревью + рефакторинг с использованием принципа single responsibility. Контроллеры больше не имеют логики обработки данных, а только отвечают на запросы, вся логика перенесена в сервис.

EvgenyAlexandrovichM committed:
10 спринт. 1 Ревью.
Исправлены недочеты по ревью + рефакторинг с использованием принципа single responsibility. Контроллеры больше не имеют логики обработки данных, а только отвечают на запросы, вся логика перенесена в сервис.
